spring-integration-core/src/main/java/org/springframework/integration/util/MessagingMethodInvokerHelper.java

```diff
@@ -831,7 +831,7 @@ private String determineHeaderExpression(Annotation headerAnnotation, MethodPara
 				headerName = methodParameter.getParameterName();
 			}
 			else if (valueAttribute.indexOf('.') != -1) {
-				String tokens[] = valueAttribute.split("\\.", 2);
+				String[] tokens = valueAttribute.split("\\.", 2);
```

Comment:
:astonished. How does that work at all.
I didn't know that Java allows this syntax :smile: